### PR TITLE
[FLINK-30819][sql-client] Fix sql client print an empty line between the multi-line statements

### PR DESCRIPTION
**Pull request by fsk119**
…tements

## What is the purpose of the change

*Fix multiline statement printing results.*

## Verifying this change

This change added tests and can be verified as follows: add tests in the SqlClientTest

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by fsk119** [comment]
@flinkbot run azure

**Comment by fsk119** [comment]
The e2e tests pass in my private pipeline:
https://dev.azure.com/1059623455/Flink/_build/results?buildId=526&view=results

**Review by LadyForest — COMMENTED** [review]
Thanks for the contribution! It looks good in general and I left some minor comments.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/Printer.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                terminal.puts(InfoCmp.Capability.clear_screen);
+            }
+        }
+
+        @Override
+        public void close() {}
+    }
+
+    /** Printer prints the initialization results. */
+    class InitializationResultPrinter implements Printer {
```
[the remark this answers is not in the document]
Comment (fsk119) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/Printer.java]: I rename it to `InitializationCommandPrinter`

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/Printer.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        public void print(Terminal terminal) {
+            printInfo(terminal, MESSAGE_EXECUTE_STATEMENT);
+        }
+
+        @Override
+        public void close() {}
+    }
+
+    /** Printer prints the statement results. */
+    class ClientResultPrinter implements Printer {
```
[the remark this answers is not in the document]
Comment (fsk119) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/Printer.java]: I rename to `StatementCommandPrinter`

**Review by LadyForest — APPROVED** [review]
Thanks for the update, LGTM

**Pull request closed without merging**